test/zdtm.py

```diff
@@ -1233,11 +1245,13 @@ def __criu_act(self, action, opts=[], log=None, nowait=False):
                     return
             rst_succeeded = os.access(
                 os.path.join(__ddir, "restore-succeeded"), os.F_OK)
-            if self.__test.blocking() or (self.__sat and action == 'restore' and
-                                          rst_succeeded):
+            if (self.__test.blocking() and ret > 0) or (self.__sat and action == 'restore' and rst_succeeded):
                 raise test_fail_expected_exc(action)
             else:
-                raise test_fail_exc("CRIU %s" % action)
+                if ((self.__criu == criu_cli and not self.__criu.signal_exit(ret)) or (self.__criu == criu_rpc and not self.__criu.signal_exit(ret))):
```

Comment:
this hunk looks strange. Why you need to check `self.__criu == criu_cli`?

Reply:
I checked it just as a precaution. Now, that I think about it, I don't think `criu_cli` would ever return a negative return code  for a non-signal exit.

Reply:
I will draft a new PR with all the changes. Closing this one.

Reply:
you don't need to create a new PR every time. You can use `git rebase -i HEAD~N`, `git commit --amend` and `git push -f` to fix your changes.

Reply:
The new PR is at https://github.com/checkpoint-restore/criu/pull/2140. Okay, I'll try not drafting a new one unnecessarily now onwards.